hivemind/utils/networking.py

```diff
@@ -1,6 +1,10 @@
 import socket
 from contextlib import closing
-from typing import Optional
+from ipaddress import ip_address
+from typing import Optional, Sequence
+
+from multiaddr import Multiaddr
+
 
 Hostname, Port = str, int  # flavour types
 Endpoint = str  # e.g. 1.2.3.4:1337 or [2a21:6с8:b192:2105]:8888, https://networkengineering.stackexchange.com/a/9435
```

Comment:
I suggest to fix this (replace `Endpoint` to `PeerID` in `dht/{node,protocol,routing}.py`) in a separate PR to avoid exploding this PR with the mechanical changes.